packages/react-tooltip/src/components/Tooltip/useTooltip.tsx

```diff
@@ -38,50 +29,50 @@ const arrowHeight = 6; // Update the arrow's width/height in useTooltipStyles.ts
  *
  * {@docCategory Tooltip}
  */
-export const useTooltip = (
-  props: TooltipProps,
-  ref: React.Ref<HTMLElement>,
-  defaultProps?: TooltipProps,
-): TooltipState => {
+export const useTooltip = (props: TooltipProps, ref: React.Ref<HTMLElement>): TooltipState => {
   const context = React.useContext(TooltipContext);
   const isServerSideRender = useIsSSR();
   const { targetDocument } = useFluent();
   const [setDelayTimeout, clearDelayTimeout] = useTimeout();
 
-  const state = mergeProps(
-    {
-      ref,
-      children: props.children,
-      content: {
-        as: React.Fragment,
-      },
-      id: useId('tooltip-'),
-      role: 'tooltip',
-      showDelay: 250,
-      hideDelay: 250,
-      triggerAriaAttribute: 'label',
-    },
-    defaultProps && resolveShorthandProps(defaultProps, tooltipShorthandProps),
-    resolveShorthandProps(props, tooltipShorthandProps),
-  );
-
-  const [visible, setVisibleInternal] = useControllableState({ state: state.visible, initialState: false });
+  const { onVisibleChange } = props;
+  const [visible, setVisibleInternal] = useControllableState({ state: props.visible, initialState: false });
   const setVisible = React.useCallback(
     (newVisible: boolean, ev?: React.PointerEvent<HTMLElement> | React.FocusEvent<HTMLElement>) => {
       clearDelayTimeout();
       setVisibleInternal(oldVisible => {
         if (newVisible !== oldVisible) {
-          const onVisibleChange = state.onVisibleChange; // Workaround for bug in react-exhaustive-deps lint rule
           onVisibleChange?.(ev, { visible: newVisible });
         }
         return newVisible;
       });
     },
-    [clearDelayTimeout, setVisibleInternal, state.onVisibleChange],
+    [clearDelayTimeout, setVisibleInternal, onVisibleChange],
   );
 
-  state.visible = visible;
-  state.shouldRenderTooltip = visible;
+  const state: TooltipState = {
+    // Default values
+    id: useId('tooltip-', props.id),
```

Comment:
This should be moved after `...props`, i.e. tsx


```diff
-id: useId('tooltip-', props.id),
...props,
+id: useId('tooltip-', props.id),
```

Otherwise a value from `props.id` will override a value from a hook.